### DISCUSSION TITLE
basedata carbon Generic in detail

### DISCUSSION THREAD

**Discussion by Yuri-HubL** [e0404/matRad#643] [Q&A]
Dear matRad administrators,

I am not sure about the entry of matRad basedata carbon_Generic data.
![base data](https://github.com/e0404/matRad/assets/72288469/e1209e1e-3487-4e06-858e-689b6017a360)

I'll list all the specific items and ask what they mean in detail (I'll use a question mark if I don't understand, hoping for an answer) :

- range: The range of the carbon ion beam of a specific energy
- energy: The energy of the monoenergetic carbon ion beam
- Depth: 135 points are taken at different depths in the beam direction
- z: (?) I don't understand the meaning of this data
- peakPos: The depth of the Bragg score
- sigma: (not sure) I think this is a record of the non-uniformity value of the beam at different depths in the beam direction
- alpha: (not sure) I see 135×2 data, which may correspond to the alpha values of tumor cells and normal tissues at different depths at different beam energies
- beta: (not sure) There are also 135×2 data, which may correspond to the beta values of tumor cells and normal tissues at different depths under different beam energies.
- offset: (?) I don't understand the meaning of this data
- initFocus: (?) I don't understand the meaning of this data
- alphaX: alpha values for normal cells and tumors under reference X-ray conditions
- betaX: beta values of normal cells and tumors under reference X-ray conditions
- alphaBetaRatio: is the ratio of alpha to beta of normal nuclear tumor cells under reference X-ray conditions
- LET: LET values at 135 different depths of the monoenergetic carbon ion beam

Looking forward to your reply.

your sincerely,
Yuri

**Comment by wahln** [comment]
Please check out our Wiki:
https://github.com/e0404/matRad/wiki/Dose-influence-matrix-calculation#particles
https://github.com/e0404/matRad/wiki/Particle-Base-Data-File
Also read the provided reference there on the pencil-beam dose calculation algorithm, Should clear up most of the stuff.

**Reply by Yuri-HubL** [reply]
Hi Wahl,

Thank you for your reply.I have some details to ask you.It mainly includes two aspects of the problem.

1. Problem of the number of outgoing particles when simulating basic data.

When I analyzed the basedata in matRad, I plotted the integrated depth dose map of 121 monoenergetic carbon ion beams according to their depth, as shown below.
![image](https://github.com/e0404/matRad/assets/72288469/d5b8d2cc-564c-45d7-ad6b-c4591d1c607a)

You can see that the higher the energy, the lower the Z-value at the incident position, and the lower the z-value at the Bragg peak. This seems to indicate that the number of particles simulated at different energies is the same when simulating the data. Am I correct? If it is correct, can anyone give me some advice on the number of outgoing particles.

2.The second aspect of the problem is the statistical location

In the plot below we can see that each point represents a data record in matRad. We can see that at different depths of the carbon ion beam the number of data to be counted is very different.
![image](https://github.com/e0404/matRad/assets/72288469/e70bfa10-58d1-4600-b9fd-b63a27c2aa11)

In the flat area, we can see that the statistical points are sparse, they become dense halfway from the flat area to the Prague peak, they become more dense at the Prague peak, and they become sparse after the Prague peak.

What I want to know is whether there is a deterministic criterion for this sparsity and denseness.

I am looking forward to your reply and that of relevant experts. Your reply will help me a lot.

your sincerely,
Yuri

**Reply by remocristoforetti** [reply]
Hi Yuri,
concerning point (1), the number of primaries simulated in the MC calculation only affects the statistical uncertainty of the outcome. The simulated Bragg Peaks are normalized in order to match the units reported in the documentation (https://github.com/e0404/matRad/wiki/Dose-influence-matrix-calculation) i.e. `[MeV cm^2 /(g * primary)]`. In this way, the stored Z profiles are normalized and thus independend on the number of primaries simulated.

Concerning point (2), the different distribution of sampling points in depth is only applied to increase the resolution in the Bragg Peak region.

Best,
Remo

**Reply by Yuri-HubL** [reply]
Hi Remo,

Thank you very much for your reply! It helped me a lot.

The simulated particle beam determines the statistical uncertainty, and I don't quite understand how this statistical uncertainty should be calculated. Are there any related web pages or articles I can refer to?

The different sampling points will affect the accuracy of matRad's calculation. Is there a good standard to determine where the sampling points should be?

Looking forward to your reply.
Yuri

**Reply by WeiWangHUST** [reply]
Hi,

I am working on building a carbon machine for calculating the biological effect using the matRad. But I don't know how to get the alpha and beta profiles with depth. I try to use the mentioned low-dose approximation method in the reference [Rapid calculation of biological effects in ion radiotherapy]  to check the default carbon machine data provided in the matRad. But, I can't get the correct results. Can you tell me how to get the alpha and beta profiles or give a detailed illustration?

Thanks.
Wei

**Reply by remocristoforetti** [reply]
Hi Wei,
thank you for your question.
Unfortunatelly there is currently no built-in code in `matRad` for explicit calculation of the alpha/beta kernels. These data depend on the specific machine configuration you are setting up and specific tissue parameters. The kernels are input data to the dose calculation algorithm and need to be provided by the user.

Please also note that there are already available open-source resources to implement the algorithm described in the reference, there is no need to write the algorithmm yourself.

Anyway, the available carbon machine provides a generic but realistic source model, unless you really need a specific machine implementation, I would suggest starting with the generic one.

Best,
Remo

**Reply by WeiWangHUST** [reply]
Hi Remo,


Thanks for your reply.


I am interested in the biological effect for ion therapy and then i want to build the specific machine based on some demands. So, can you share me the related open-source resources to implement the algorithm.
Thanks again.


Best,
Wei

**Reply by saxu3829** [reply]
Hi Remo,

I am also very intersted which open-source resources there are to implement the algorithm since I have not found any. Can you share your insight on this topic? 

Best,
Jessica

**Reply by wahln** [reply]
There is an open survival code to compute alpha / beta tables (and corresponding quantities) for use in simulations:
https://github.com/batuff/Survival

You can then use, for example, the tabulated RBE scorers in TOPAS to calculate ion-specific alpha and beta and extract those for the kernels for integration in matRad.

In the future, we hope to include a more informative kernel dataset which will be able to use such RBE tables directly (by containing fluence/energy spectra).

**Reply by saxu3829** [reply]
Thank you for your reply! I have another question regarding the base data that is unfortunately not covered by the wiki. Which unit is used for the LET in the base data?

**Reply by wahln** [reply]
The LET should be keV/micrometer. How you score it for the database is your choice (so you can store either track-averaged or dose-averaged, restricted or unrestricted), but beware that matRad will then use it to compute a dose-weighted LET distribution (more exactly, matRad will multiply LET * dose for each spot for each voxel, then accumulate LET * dose over all spots per voxel, and finally divide by the total dose in each voxel).

**Reply by saxu3829** [reply]
Thank you for your quick reply! I looked into the following paper to extract the Monte Carlo simulation settings for the creation of base data for a clinical particle therapy facility: K Parodi et al 2012 Phys. Med. Biol. 57 3759. It was cited by the matRad paper on the base data for protons and carbons as well.
If I understood the paper correctly, all simulations for the ddd were performed without the nozzle. For the matRad base data, my first suggestion would have been to include the nozzle in the ddd and fragmentation spectra simulations to match the data to the respective facility. Am I wrong for thinking that way?
Another thing that I did not understand was which type of geometry was used for the laterally integrated ddd. There were two geometries used, one cylinder with a diameter of 28 cm and one with a diameter of the active volume of an IC. Since the volume used to score the laterally integrated ddd plays an important role on the output dose, it would be nice if you could share your insights with me.

**Reply by wahln** [reply]
If I remember correctly, the work of K Parodi wanted to include nozzle composition but separate from the air. One of the reasons probably were similar nozzles in all rooms but different nozzle-to-isocenter distances (and, secondary, different patient geometries). So I think what was done is the paper is to simulate the nozzle geometry (there was some trickery with approximating the BAMS), but then assume vacuum. 

matRad doesn't bind you to make the same decisions. You can just simulate with your nozzle configuration / your phase space behind the nozzle and then store the ddd. The few fields that steer corrections due to simulation differences (e.g., every field has an "offset" field that allows to describe a constant difference in WEPL for this entry) could then be set to not correct (e.g. offset = 0).

When we generate new ddds, we usually just create a "large enough" water cylinder binned in depth.
I think the two geometries were mainly used by Parodi et al. to compare measurements with simulations and check their consistency.

If you do some simulation now, store ouputs well. We are currently implementing/revising the storage of spectra etc. for more flexible biological modeling and forwarding to MC. 

One thing already present on the "dev" branch is that you can store now a energy spectrum parameterization at nozzle exit in machine.data.energySpectrum (a struct with the fields "mean" for mean energy (in MeV/u for carbon) and "sigma" in % for energy spectrum width). Similarly, machine.data.initFocus can hold a struct array called "emittance" with fields sigmaX, divX, corrX (and Y) to store a bigaussian emittance parameterization for each focus).
When using a Monte Carlo dose calculation engine, matRad uses those for the simulations. if not present, it tries to reverse engineer the beam optics and energy. Check this dataset, for example, https://github.com/e0404/matRad/blob/dev/matRad/basedata/protons_generic_TOPAS.mat

This is also helpful for quick consistency checks between a pencil-beam dataset and Monte Carlo.

**Reply by Yuri-HubL** [reply]
Hi wahln，

I would like to ask, during simulation, are ### water phantoms used? 
What kind of ripple filters used to broaden the Bragg peak of carbon ions beam? 
Is there somewhere I can refer to?

have a nice day!
yuri

**Reply by saxu3829** [reply]
Hi, 

I was going through the different parameters needed for the base data and am not quite sure if I understood this correctly in the wiki: Where is the zero value of 'dist' in machine.data.initFocus? In my opinion it should be at the point of the source. But if you only need it to model the beam widening of the beam nozzle to the patient surface, then it could also be at the end of the beam nozzle. 

Thank you in advance!
Best regards,
Jessica

**Reply by wahln** [reply]
The `dist` value is relative to the (virtual) scanning source, so having a value at 0 would not make too much sense. That's also why the values are all relatively large. Subtract the `SAD` to make them relative to the isocenter, for example. To make them relative to the nozzle, use `BAMStoIsoDist` (strictly speaking that is the distance from the Monitoring System to the IsoCenter, but we use it as the nozzle position).

**Reply by saxu3829** [reply]
Thank you for your quick reply! If I understand correctly, you could for example say if the sigma at the end of the beam nozzle is X, then dist = SAD-BAMStoIsoDist, right? And we then could approximate all values before this point with the same sigma, so that sigma[0, SAD-BAMStoIsoDist] = X? Or is it also correct if you don't define a sigma value between zero and SAD-BAMStoIsoDist?

**Reply by wahln** [reply]
you don't need to define a sigma before the nozzle. What matRad is doing with this data is to interpolate the initial width of the beam when impinging on the patient: we compute the SSD, which should always be > BAMStoIsoDist, and then interpolate at the SSD value in dist & sigma to get the initial beam sigma used to convolve the lateral kernel with.

**Reply by ni484call5** [reply]
Hello!

I notice that multi gaussian seems supported in the code
https://github.com/e0404/matRad/blob/ca05aed22762deed3b0d54289bda5c2ab8d96660/matRad/doseCalc/%2BDoseEngines/matRad_ParticlePencilBeamEngineAbstract.m#L224C1-L237C16

I'm wonder is **triple gaussian** supported or not in carbon? If so, what's the format should be in carbon_Generic for `baseData.weightMulti` and `baseData.sigmaMulti`?

Sincerely,
Nicholas

**Reply by wahln** [reply]
@ni484call5 yes, in the new version you can have a triple-gaussian model using these entreis (they are not present in the carbon_Generic data, though).
- `sigmaMulti`: needs to be a d x n matrix where d is the length of the `depths` vector and n is the number of Gaussian components. Thus each kernel component is tabulated in depth at the same points, with components' data stacked in columns
- `weightMulti`: needs to be a d x (n-1) matrix representing the weights in depth of the 2nd to last component per column. The weight of the first component will be calculated from the condition that all components weights must sum up to 1.
